### ISSUE TITLE
Track keyword comment in the new control flow leads to a broken link 

### ISSUE THREAD

**Issue by ViktorTyultinov** [angular/angular#56410]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

While working with the new control flow I wanted to check the track keyword documentation. The link to the documentation in the hover leads to https://next.angular.io/api/core/for instead of https://angular.dev/guide/templates/control-flow#track-for-calculating-difference-of-two-collections 

P.S After a quick check I found that most quick info links are broken in the same way.

### Please provide the exception or error you saw

```true
https://next.angular.io/api/core/for
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 18.0.3
Node: 20.10.0
Package Manager: npm 10.5.1
OS: win32 x64

Angular: 18.0.2
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1800.3
@angular-devkit/build-angular   18.0.3
@angular-devkit/core            18.0.3
@angular-devkit/schematics      18.0.3
@angular/cli                    18.0.3
@schematics/angular             18.0.3
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.2
```

**Comment by ViktorTyultinov** [comment]
I would love to fix this if possible :)

**Comment by JeanMeche** [comment]
If you're indeed talking about the language service which provides those links, you can open a PR which changes on this file. 
https://github.com/angular/angular/blob/main/packages/language-service/src/quick_info_built_ins.ts#L140

**Comment by ViktorTyultinov** [comment]
@JeanMeche Yes this is exactly what I am talking about. I didn't find suitable links for all of the descriptions so I have mostly redirected it to https://angular.dev/guide/defer#defer

**Comment by alxhub** [comment]
Fixed by #56416 - thanks @ViktorTyultinov!

**Comment by ViktorTyultinov** [comment]
I am very happy I could help!
